routers/web/repo/actions/view.go

```diff
@@ -304,6 +306,42 @@ func ViewPost(ctx *context_module.Context) {
 	ctx.JSON(http.StatusOK, resp)
 }
 
+// getWorkflowFileLink return url for the source of the workflow file that was run
+func getWorkflowFileLink(ctx *context_module.Context, run *actions_model.ActionRun) string {
+	if run.Repo == nil || run.CommitSHA == "" {
+		return ""
+	}
+
+	commit, err := ctx.Repo.GitRepo.GetCommit(run.CommitSHA)
+	if err != nil {
+		return ""
+	}
+
+	entries, err := actions.ListWorkflows(commit)
+	if err != nil {
+		return ""
+	}
+
+	var workflowEntry *git.TreeEntry
+	for _, entry := range entries {
+		if entry.Name() == run.WorkflowID {
```

Comment:
Wait, we are relying on the workflow ID being the same as the filename?
This sounds like technical debt waiting to happen.
Do we not store which workflow triggered a run in some other way?

Reply:
Yeah, that seems to be the canonnical way it's handled in the code-base right now. I think a better solution would be to generate some uuid, then add a data object for the path to the workflow file, but that's probably out of scope for this PR, which is just building a link to the file.